### PR TITLE
Docker: bump package-spec version to 3.2.2

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Bump package-spec version to 3.2.2.

Reasons:

- Align with best practices and requirements in package-spec v3.
- Serverless requires a package-spec version of 3.0.x.

Notes: 

- As of today, to test the Docker integration on non-serverless stack version 9.0, you must build the package with a `conditions.kibana.version: >= 8.8.0`.
- Future elastic-package version https://github.com/elastic/elastic-package/pull/2116 will allow test the integration on stack version `9.0.0-SNAPSHOT` without changing `conditions.kibana.version`.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

-

### CODE REVIEW

**Comment by zmoog** [comment]
I'm not 100% sure about `object_type` for the `docker.memory.stats.*` fields.

File: packages/docker/data_stream/memory/fields/fields.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@
   fields:
     - name: stats.*
       type: object
+      object_type: float
```
Comment (MichaelKatsoulis) [inline packages/docker/data_stream/memory/fields/fields.yml:12]: I think the correct would be `long` or `double`. But looking at the stats, I would pick `long`

**Comment by gizas** [comment]
LGTM, package build is successful without any errors

**Review by gizas — APPROVED** [review]

**Comment by zmoog** [comment]
Tested on stack version 9.0.0-SNAPSHOT and package-spec 3.2.2:

```
$ docker images | grep 9.0.0-SNAPSHOT

docker.elastic.co/elastic-agent/elastic-agent-wolfi      9.0.0-SNAPSHOT   6a86b26fe188   3 days ago      1.58GB
docker.elastic.co/elasticsearch/elasticsearch            9.0.0-SNAPSHOT   535cb5782b83   3 days ago      1.38GB
docker.elastic.co/kibana/kibana                          9.0.0-SNAPSHOT   277a97ad4979   4 days ago      1.93GB
```

And I can successfully see and install the Docker integration:

https://github.com/user-attachments/assets/2e5c6c67-4563-47ae-93c5-af872cf0d6a4

**Comment by zmoog** [comment]
@gizas @MichaelKatsoulis, could you take a final look when you have time?

**Comment by MichaelKatsoulis** [comment]
> @gizas @MichaelKatsoulis, could you take a final look when you have time?

LGTM!

**Comment by gizas** [comment]
all good @zmoog from side!

**Pull request merged**